### PR TITLE
`Adaptive learning`: Add success and warning alerts to competency generation

### PR DESCRIPTION
**Pull request by rstief**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
When using competency generation with very short descriptions (or when testing with nonsense), sometimes no competencies were generated to the LLM being unable to extract any from the text. This was confusing to the user, as no success/error text was displayed.

### Description
- Adds success/warning alerts to competency generation
- Prevents the user from accidentally reloading the page when using competency generation 

### Steps for Testing
Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Management -> Competencies -> Generate Competencies
3. Enter a nonsense course description ("asdsadasdasdsa...")
4. Click on `Generate`
5. See that a warning gets displayed that no competencies could be generated
6. Retry with a "meaningful" course description (e.g. [advanced_topics_of_se.txt](https://github.com/ls1intum/Artemis/files/14397446/advanced_topics_of_se.txt))
7. See that competency recommendations are generated an a success message is displayed
8. Try to reload the page -> See that there is a warning before

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [generate-competencies.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6128/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/competencies/generate-competencies/generate-competencies.component.ts.html) | 96.96% | ✅  |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [IrisCompetencyGenerationSessionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS6128/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.iris.session/IrisCompetencyGenerationSessionService.html) | 87% | ✅  |

## Summary by CodeRabbit

- **New Features**
	- Updated the logic for skipping competencies in the competency generation service to enhance user experience.
	- Improved the course description form layout with minor styling adjustments.
	- Enhanced the generate competencies component with additional spacing for better readability.
	- Integrated translation services to support multilingual capabilities.
	- Implemented a new warning system to alert users of unsaved changes before navigating away or refreshing the page.

### CODE REVIEW

**Review by milljoniaer — DISMISSED** [review]
tested in testing session, works as expected

**Review by laurenzfb — DISMISSED** [review]
demonstrated in testing session, works

**Review by basak-akan — DISMISSED** [review]
Code looks good 👍

**Review by reschandreas — DISMISSED** [review]
code lgtm

**Review by MaximilianAnzinger — APPROVED** [review]

File: src/main/webapp/app/course/competencies/generate-competencies/generate-competencies.component.ts [hunk 1]
```diff
@@ -170,4 +170,25 @@ export class GenerateCompetenciesComponent implements OnInit, ComponentCanDeacti
     get competencies() {
         return this.form.controls.competencies;
     }
+
+    /**
+     * Only allow to leave page after submitting or if no pending changes exist
+     */
+    canDeactivate(): boolean {
+        return this.submitted || (!this.isLoading && !(this.competencies.length > 0));
```
Comment (bassner) [inline src/main/webapp/app/course/competencies/generate-competencies/generate-competencies.component.ts:178]:
```suggestion
        return this.submitted || (!this.isLoading && this.competencies.length === 0);
```

**Review by bassner — CHANGES_REQUESTED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]

**Review by laurenzfb — APPROVED** [review]

**Review by basak-akan — APPROVED** [review]
Reapprove

**Review by reschandreas — APPROVED** [review]
reapprove

**Review by bassner — APPROVED** [review]

**Pull request merged**